src/rpc/rawtransaction.cpp

```diff
@@ -1210,45 +1241,114 @@ static UniValue testmempoolaccept(const JSONRPCRequest& request)
     }
 
     RPCTypeCheck(request.params, {UniValue::VARR, UniValue::VBOOL});
-    if (request.params[0].get_array().size() != 1) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Array must contain exactly one raw transaction for now");
+    const UniValue raw_transactions = request.params[0].get_array();
+    if (raw_transactions.size() < 1 || raw_transactions.size() > MAX_PACKAGE_COUNT) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                            "Too many transactions in package." );
     }
 
-    CMutableTransaction mtx;
-    if (!DecodeHexTx(mtx, request.params[0].get_array()[0].get_str())) {
-        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+    Package package;
+    for (const auto& rawtx : raw_transactions.getValues()) {
+        CMutableTransaction mtx;
+        if (!DecodeHexTx(mtx, rawtx.get_str())) {
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR,
+                                "TX decode failed: " + rawtx.get_str() + " Make sure the tx has at least one input.");
+        }
+        package.push_back(MakeTransactionRef(std::move(mtx)));
+    }
+
+    if(package.size() != raw_transactions.size()){
+            throw JSONRPCError(RPC_INVALID_PARAMETER,
+                    "Duplicate transaction found in package.");
     }
-    CTransactionRef tx(MakeTransactionRef(std::move(mtx)));
-    const uint256& tx_hash = tx->GetHashMalFix();
 
     CAmount max_raw_tx_fee = ::maxTxFee;
     if (!request.params[1].isNull() && request.params[1].get_bool()) {
         max_raw_tx_fee = 0;
     }
 
-    UniValue result(UniValue::VARR);
-    UniValue result_0(UniValue::VOBJ);
-    result_0.pushKV("txid", tx_hash.GetHex());
+    CValidationState state;
+    PackageValidationState pkg_results;
+    std::vector<CTxMemPoolEntry> submitPool;
+    submitPool.reserve(package.size());
 
-    CTxMempoolAcceptanceOptions opt;
-    opt.flags = MempoolAcceptanceFlags::TEST_ONLY;
-    bool accept;
-    {
-        LOCK(cs_main);
-        accept = AcceptToMemoryPool(std::move(tx), opt);
-    }
-    result_0.pushKV("allowed", accept);
-    if (!accept) {
-        if (opt.state.IsInvalid()) {
-            result_0.pushKV("reject-reason", strprintf("%i: %s", opt.state.GetRejectCode(), opt.state.GetRejectReason()));
-        } else if (opt.missingInputs.size()) {
-            result_0.pushKV("reject-reason", "missing-inputs");
-        } else {
-            result_0.pushKV("reject-reason", opt.state.GetRejectReason());
+    bool success = false;
+    success = TestPackageAcceptance(package, state, pkg_results, &submitPool);
+
+    UniValue result(UniValue::VOBJ);
+
+    encodePackageResult(success, pkg_results, state, result);
+
+    return result;
+}
+
+static UniValue submitpackage(const JSONRPCRequest& request)
```

Comment:
Just like Bitcoin, move to `rpc/mempoo.cpp`. `testmempoolaccept` too.

Reply:
Bitcoin's mempool.cpp has all mempool related rpcs. I'll add only testmempoolaccept and submitpackage in to mempool.cpp in this PR. move the other rpcs in separate PR.